zio/detector/file.go

```diff
@@ -61,6 +62,25 @@ func OpenFile(zctx *resolver.Context, path string, cfg OpenConfig) (*zbuf.File,
 	return OpenFromNamedReadCloser(zctx, f, path, cfg)
 }
 
+type checkReader struct {
+	r zbuf.Reader
+}
+
+func (c *checkReader) Read() (*zng.Record, error) {
+	rec, err := c.r.Read()
+	if rec == nil || err != nil {
+		return rec, err
+	}
+	if err := rec.TypeCheck(); err != nil {
+		return nil, err
+	}
+	return rec, nil
+}
+
+func newCheckReader(r zbuf.Reader) zbuf.Reader {
```

Comment:
This function isn't used, and I think it's fine to create a checkReader with just `&checkReader{r}`, so suggest deleting it.

Reply:
Oops it's a leftover. Will remove.